packages/core/lib/Bus.ts

```diff
@@ -77,6 +109,19 @@ export class Bus<A extends Actor<I, T, O>, I extends IAction, T extends IActorTe
     });
   }
 
+  /**
+   * Invoked when an action was run by an actor.
+   *
+   * @param actor               The action on which the {@link Actor#run} method was invoked.
+   * @param {I}          action The original action input.
+   * @param {Promise<O>} output A promise resolving to the final action output.
+   */
+  public onRun(actor: Actor<I, T, O>, action: I, output: Promise<O>): void {
+    for (const observer of this.observers) {
+      observer.onRun(actor, action, output);
```

Comment:
It is possible that an observer has an async onRun function. In that case, is it not possible that a stream in the output object is already drained before it reaches the observer (should the calling actor receive it first)?

Reply:
In the case of streams, I would say that the observer implementor is responsible for making sure that it doesn't mess with the actor's handling. For non-streams, I don't think it's a problem if onRun is async.

Reply:
I meant the reverse. The actor might have already read some things from the stream before the observer actually gets access to the stream, causing it to not see everything that was in the output